streams/src/main/java/org/apache/kafka/streams/processor/internals/ActiveTaskCreator.java

```diff
@@ -79,7 +83,7 @@ private static String getTaskProducerClientId(final String threadClientId, final
                       final Time time,
                       final KafkaClientSupplier clientSupplier,
                       final String threadId,
-                      final UUID processId,
+                      final String eosBetaTxIdSuffix,
```

Comment:
Passing in the `processId` does not create unique `transaction.id` for all threads, because the `processId` is unique per `KafkaStreams` instance and the same over all threads within an instance. Thus, I propose to use the following `transactional.id` instead:
 ```
applicationId + "-" + processId + "-StreamThread-" + threadIdx
```

This variable is the corresponding suffix `processId + "-StreamThread-" + threadIdx`

Reply:
No, but I am also not sure how to test this in a unit test (suggestions are welcome)? It was caught by an integration test so I was hoping we are fine?

Reply:
We could do that, but does it make the code any simpler?

Reply:
I think it is somewhat subjective :) My rationale is that it's better to not pass in parameters for a specific mode into the task creator which may not be used at all; i.e. we encapsulate the logic of creating transaction.id when eos-alpha / beta is used only inside the callee (the constructor) rather than letting the caller to prepare for it without knowing if it would really be needed or not.